quinn-proto/src/tests/mod.rs

```diff
@@ -1780,6 +1780,59 @@ fn congested_tail_loss() {
     pair.client_send(client_ch, s).write(&[42; 1024]).unwrap();
 }
 
+// Send a tail-loss probe when GSO segment_size is less than INITIAL_MTU
+#[test]
+fn tail_loss_small_segment_size() {
+    let _guard = subscribe();
+    let mut pair = Pair::default();
+    let (client_ch, server_ch) = pair.connect();
+
+    // No datagrams frames received in the handshake.
+    let server_stats = pair.server_conn_mut(server_ch).stats();
+    assert_eq!(server_stats.frame_rx.datagram, 0);
+
+    const DGRAM_LEN: usize = 1000; // Below INITIAL_MTU after packet overhead.
+    const DGRAM_NUM: usize = 5; // Enough to build a GSO batch.
+
+    info!("Sending 1st datagram batch");
+    for _ in 0..DGRAM_NUM {
+        pair.client_datagrams(client_ch)
+            .send(vec![0; DGRAM_LEN].into(), false)
+            .unwrap();
+    }
+    pair.drive_client();
```

Comment:
Does this flight need to have any particular structure, or could any ACK-eliciting packet (e.g. a single ping) do the trick?

Reply:
Indeed, any ack-eliciting packet is fine.  I went with sending the same twice but maybe ack is even more clear.  Changed this.